frontend/app/components/wp-relations/wp-relation-group/wp-parent-relation-group.service.ts

```diff
@@ -46,24 +56,39 @@ export class WorkPackageParentRelationGroup extends WorkPackageRelationGroup {
   }
 
   public addWpRelation(wpId:number) {
-    this.workPackage.parentId = wpId;
-
-    return this.workPackage.save().then(() => {
+    return this.changeParent(wpId).then(() => {
       this.workPackage.parent.$load().then(parent => {
         this.relations[0] = parent;
       });
     });
   }
 
   public removeWpRelation() {
-    this.workPackage.parentId = null;
-
-    return this.workPackage.save().then(() => {
+    return this.changeParent(null).then(() => {
       this.relations.pop();
       return 0;
     });
   }
 
+  protected changeParent(parentId:number) {
+    var params = {
+      parentId: parentId,
+      lockVersion: this.workPackage.lockVersion
+    };
+
+    return this.workPackage.changeParent(params)
+      .then((wp) => {
+        this.workPackage = wp;
+        return wpCacheService.updateWorkPackage(wp);
+      })
+      .catch(error => {
+        if (!(error.data instanceof ErrorResource)) {
```

Comment:
Shouldn't this be an if-else where you only show the specific error if we have an ErrorResource here?